src/serializer.rs

```diff
@@ -1,75 +1,112 @@
-use crate::parser::*;
-use bytes::{BufMut, BytesMut};
+use std::io::{self, Write};
 
-/// Serialization of a SomeIP header to bytes
-#[allow(dead_code)]
-fn serialize_someip_header(header: &SomeIpHeader) -> Vec<u8> {
-    let mut buf = BytesMut::with_capacity(144);
-    buf.put_u16(header.message_id.service_id.into());
-    buf.put_u16(header.message_id.method_id.into());
-    buf.put_u32(header.length);
-    buf.put_u16(header.request_id.client_id.into());
-    buf.put_u16(header.request_id.session_id.into());
-    buf.put_u8(header.protocol_version);
-    buf.put_u8(header.interface_version);
-    buf.put_u8(header.message_type.into());
-    buf.put_u8(header.return_code.into());
+use byteorder::{BigEndian, WriteBytesExt};
+
+use crate::types::{Header, MessageID, MessageType, RequestID, ReturnCode, SomeIp};
+
+/// Header length in bytes PRS_SOMEIP_00030
+const HEADER_LENGTH: usize = 16;
```

Comment:
Duplicates HEADER_LENGTH in parser.rs

Reply:
Yes. Intentional because there's no better place for it. This number will never change.